### PR TITLE
Homework 2. Attempt 1

### PR DESCRIPTION
**Pull request by OGrabar**

### CODE REVIEW

File: src/test/java/hw2/TestBase.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    @AfterClass
+    public void afterClass() {
+        driver.close();
+    }
+
+    protected static void checkBrowserTitle() {
+        softAssert.assertEquals(driver.getTitle(), EXPECTED_HOME_PAGE_TITLE);
+    }
+
+    protected static void performLogin() {
```
Comment (HronoSF) [inline src/test/java/hw2/TestBase.java]: Do you know about enums?
I want you to create enum LoginUser with fields: login, password, username. Then method login like: protected void login(LoginUser loginUser){.....}

File: src/test/java/hw2/TestBase.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    private static final String DEFAULT_LOGIN = "Roman";
+    private static final String DEFAULT_PASSWORD = "Jdi1234";
+    private static final String DEFAULT_USERNAME = "ROMAN IOVLEV";
+    private static final String LOGIN = System.getProperty("login", DEFAULT_LOGIN);
+    private static final String PASSWORD = System.getProperty("password", DEFAULT_PASSWORD);
+    private static final String USERNAME = System.getProperty("user", DEFAULT_USERNAME);
+
+    private static final String EXPECTED_HOME_PAGE_TITLE = "Home Page";
+
+    private static final String LOGIN_DROPDOWN_CLASS_SELECTOR = "uui-profile-menu";
```
Comment (HronoSF) [inline src/test/java/hw2/TestBase.java]: Do not create constants for locators, use in directly in method calls, remove all of them

File: src/test/java/hw2/TestBase.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.asserts.SoftAssert;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class TestBase {
```
Comment (HronoSF) [inline src/test/java/hw2/TestBase.java]: Class with "Base" in name should has abstract modifier according to java convention

File: src/test/java/hw2/TestBase.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    private static final String DEFAULT_URL = "https://jdi-testing.github.io/jdi-light/index.html";
+    private static final String URL = System.getProperty("url", DEFAULT_URL);
+
+    private static final String DEFAULT_LOGIN = "Roman";
+    private static final String DEFAULT_PASSWORD = "Jdi1234";
```
Comment (HronoSF) [inline src/test/java/hw2/TestBase.java]: I want you to create enum LoginUser with fields: login, password, username

File: src/test/java/hw2/FirstExerciseTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    }};
+    private static final List<String> EXPECTED_SIDE_NAV_BAR_ITEMS_TEXT = new ArrayList<String>() {{
+        add("Home");
+        add("Contact form");
+        add("Service");
+        add("Metals & Colors");
+        add("Elements packs");
+    }};
+
+    private static final String SIDE_NAVIGATION_BAR_NAME_SELECTOR = "navigation-sidebar";
```
Comment (HronoSF) [inline src/test/java/hw2/FirstExerciseTest.java]: Do not create constants for locators, use in directly in method calls, remove all of them

File: src/test/java/hw2/FirstExerciseTest.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        add("To be multiplatform");
+        add("Already have good base\n(about 20 internal and\nsome external projects),\nwish to get more…");
+    }};
+    private static final List<String> EXPECTED_HEADER_ITEMS_TEXT = new ArrayList<String>() {{
+        add("HOME");
+        add("CONTACT FORM");
+        add("SERVICE");
+        add("METALS & COLORS");
+    }};
+    private static final List<String> EXPECTED_SIDE_NAV_BAR_ITEMS_TEXT = new ArrayList<String>() {{
```
Comment (HronoSF) [inline src/test/java/hw2/FirstExerciseTest.java:26]: move to enum

File: src/test/java/hw2/FirstExerciseTest.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FirstExerciseTest extends TestBase {
+
+    private static final int EXPECTED_BENEFITS_COUNT = 4;
+    private static final List<String> EXPECTED_BENEFIT_TEXTS = new ArrayList<String>() {{
+        add("To include good practices\nand ideas from successful\nEPAM project");
```
Comment (HronoSF) [inline src/test/java/hw2/FirstExerciseTest.java:15]: move to enum

File: src/test/java/hw2/FirstExerciseTest.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+        add("To be flexible and\ncustomizable");
+        add("To be multiplatform");
+        add("Already have good base\n(about 20 internal and\nsome external projects),\nwish to get more…");
+    }};
+    private static final List<String> EXPECTED_HEADER_ITEMS_TEXT = new ArrayList<String>() {{
```
Comment (HronoSF) [inline src/test/java/hw2/FirstExerciseTest.java:20]: move to enum

File: src/test/java/hw2/FirstExerciseTest.java [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FirstExerciseTest extends TestBase {
+
+    private static final int EXPECTED_BENEFITS_COUNT = 4;
```
Comment (HronoSF) [inline src/test/java/hw2/FirstExerciseTest.java:13]: use directly in test

File: src/test/java/hw2/SecondExerciseTest.java [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+import java.util.List;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.Test;
+
+public class SecondExerciseTest extends TestBase{
+
+    private static final String EXPECTED_DIFFERENT_ELEMENTS_PAGE_TITLE = "Different Elements";
+    private static final String WIND_CHECKBOX = "Wind";
```
Comment (HronoSF) [inline src/test/java/hw2/SecondExerciseTest.java:11]: Do not create constants for locators, use in directly in method calls, remove all of them

File: src/test/java/hw2/SecondExerciseTest.java [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+        //8. Select in dropdown
+        selectDropdownOption(YELLOW_DROPDOWN_OPTION);
+
+        //9. Assert logs
+        checkLogs();
+
+        softAssert.assertAll();
+    }
+
+    private static void goToDifferentElementsPage() {
```
Comment (HronoSF) [inline src/test/java/hw2/SecondExerciseTest.java]: why static?

File: src/test/java/hw2/SecondExerciseTest.java [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+        WebElement topNavigationBarServiceItem = driver.findElement(By.xpath(SERVICE_MENU_XPATH_SELECTOR));
+        topNavigationBarServiceItem.click();
+        WebElement serviceMenuDifferentElementItem = driver.findElement(By.xpath(SERVICE_MENU_DIFFERENT_ELEMENTS_XPATH_SELECTOR));
+        serviceMenuDifferentElementItem.click();
+        softAssert.assertEquals(driver.getTitle(), EXPECTED_DIFFERENT_ELEMENTS_PAGE_TITLE);
+    }
+
+    private static void clickCheckBox(String checkBox) {
```
Comment (HronoSF) [inline src/test/java/hw2/SecondExerciseTest.java]: why static?

File: src/test/java/hw2/SecondExerciseTest.java [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+        driver.findElement(By.xpath(String.format(CHECKBOX_XPATH_SELECTOR, checkBox))).click();
+    }
+
+    private static void clickRadioButton(String radioButtonText) {
+        driver.findElement(By.xpath(String.format(RADIO_BUTTON_XPATH_SELECTOR, radioButtonText))).click();
+    }
+
+    private static void selectDropdownOption(String dropdownOption) {
```
Comment (HronoSF) [inline src/test/java/hw2/SecondExerciseTest.java]: why static?

File: src/test/java/hw2/SecondExerciseTest.java [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+       driver.findElement(By.xpath(String.format(DROPDOWN_OPTION_XPATH_SELECTOR, dropdownOption))).click();
+    }
+
+    private static void checkLogs() {
```
Comment (HronoSF) [inline src/test/java/hw2/SecondExerciseTest.java]: why static?

File: src/test/java/hw2/SecondExerciseTest.java [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+        List<WebElement> logsItems = driver.findElements(By.xpath(LOGS_XPATH_SELECTOR));
+        List<String> logsText = getNotEmptyTextFromWebElements(logsItems);
+
+        softAssert.assertTrue(logsText.get(logsText.size() - 1).contains(WATER_CHECKBOX));
```
Comment (HronoSF) [inline src/test/java/hw2/SecondExerciseTest.java]: logsText.size() - 1 and etc. is hard.
Simplify pls

File: src/test/java/hw2/TestBase.java (hunk 1 above)
Comment (OGrabar) [reply src/test/java/hw2/TestBase.java]: done

File: src/test/java/hw2/TestBase.java (hunk 2 above)
Comment (OGrabar) [reply src/test/java/hw2/TestBase.java]: done

File: src/test/java/hw2/TestBase.java (hunk 3 above)
Comment (OGrabar) [reply src/test/java/hw2/TestBase.java]: done

File: src/test/java/hw2/TestBase.java (hunk 4 above)
Comment (OGrabar) [reply src/test/java/hw2/TestBase.java]: done

File: src/test/java/hw2/FirstExerciseTest.java (hunk 5 above)
Comment (OGrabar) [reply src/test/java/hw2/FirstExerciseTest.java]: done

File: src/test/java/hw2/SecondExerciseTest.java (hunk 11 above)
Comment (OGrabar) [reply src/test/java/hw2/SecondExerciseTest.java]: fixed

File: src/test/java/hw2/SecondExerciseTest.java (hunk 12 above)
Comment (OGrabar) [reply src/test/java/hw2/SecondExerciseTest.java]: fixed

File: src/test/java/hw2/SecondExerciseTest.java (hunk 13 above)
Comment (OGrabar) [reply src/test/java/hw2/SecondExerciseTest.java]: fixed

File: src/test/java/hw2/SecondExerciseTest.java (hunk 14 above)
Comment (OGrabar) [reply src/test/java/hw2/SecondExerciseTest.java]: fixed

File: src/test/java/hw2/SecondExerciseTest.java (hunk 15 above)
Comment (OGrabar) [reply src/test/java/hw2/SecondExerciseTest.java]: fixed

File: pom.xml [hunk 16]
```diff
@@ -8,6 +8,63 @@
     <artifactId>Oleg.Grabar</artifactId>
     <version>1.0-SNAPSHOT</version>
 
+    <properties>
```
Comment (HronoSF) [inline pom.xml:11]: fix merge conflict with master!

**Pull request merged**